### PR TITLE
Shooter code

### PR DESCRIPTION
**Pull request by ProgrammingSR**

### CODE REVIEW

File: src/main/java/frc/robot/HardwareMap.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj.SpeedController;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.XboxController;
+
+public class HardwareMap {
+
+    public class ShooterHardware{
+        public CANSparkMax leftShooter = new CANSparkMax(16, MotorType.kBrushless);
```
Comment (noam987) [inline src/main/java/frc/robot/HardwareMap.java]: move the construction of objects into the constructor of shooter hardware. (the new CANSparkmax part)

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  private CANSparkMax m_leftShooter;
+  private CANSparkMax m_rightShooter;
+  private SpeedControllerGroup m_shooter;
+  private double m_targetSpeed;
+  /** Creates a new ShooterSubsystem. */
+  public ShooterSubsystem(ShooterHardware shooter) {
+    m_leftShooter = shooter.leftShooter;
+    m_rightShooter = shooter.rightShooter;
+  
+    m_shooter = shooter.shooter;
```
Comment (noam987) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java:25]: why are you taking in both motors if you don't use them.

**Review by noam987 — CHANGES_REQUESTED** [review]
2 simple minor changes to clean it up, but overall great

**Comment by ProgrammingSR** [comment]
Needs to work on bang bang loop

**Comment by noam987** [comment]
Consider doing it in two parts. Get the base code in first, add bang bang in the next PR

**Review by noam987 — APPROVED** [review]
I approve but have someone else also review it. Does Shoot one ball command need to be merged now?

**Review by noam987 — APPROVED** [review]

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.HardwareMap;
+import frc.robot.HardwareMap.ShooterHardware;
+
+public class ShooterSubsystem extends SubsystemBase {
+  private SpeedControllerGroup m_shooter;
+  private SpeedControllerGroup m_feeder;
+  private CANEncoder m_canEncoder;
```
Comment (thetaspirit) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: Please rename this

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  /** Creates a new ShooterSubsystem. */
+  public ShooterSubsystem(ShooterHardware shooter) {    
+    m_shooter = shooter.shooter;
+    m_feeder = shooter.feeder;
+    m_canEncoder = shooter.rightCanEncoder;
+    
+
+  }
+  public void setShooter(double speed){
```
Comment (thetaspirit) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java:30]: The following methods need javadocs.  Writing a javadoc for `periodic` is up to you.

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+public class ShooterSubsystem extends SubsystemBase {
+  private SpeedControllerGroup m_shooter;
+  private SpeedControllerGroup m_feeder;
+  private CANEncoder m_canEncoder;
+  private double m_targetSpeed;
+  private double m_feederSpeed;
+  
+
+  /** Creates a new ShooterSubsystem. */
+  public ShooterSubsystem(ShooterHardware shooter) {
```
Comment (thetaspirit) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java:23]: Please also briefly explain the distinction between the shooter and feeder in javadocs for this constructor.  Also rename `m_canEncoder` and briefly describe its purpose.

**Review by thetaspirit — CHANGES_REQUESTED** [review]

**Review by TiaSinghania — APPROVED** [review]
javadocs need to be added but that can be done with a seperate PR

**Pull request merged**